core/store/src/trie/ops/iter.rs

```diff
@@ -95,9 +95,12 @@ where
     I: GenericTrieInternalStorage<N, V>,
 {
     /// Create a new iterator.
+    /// If `start` is specified, the iterator will be positioned on the first element
+    /// with key >= `start`, without recording nodes accessed during the seek.
     pub fn new(
         trie_interface: I,
         prune_condition: Option<Box<dyn Fn(&Vec<u8>) -> bool>>,
+        start: Option<Vec<u8>>,
```

Comment:
As discussed offline, we can try to not include this in the new function and instead not call `descend_into_node` on root to prevent us from unintentionally recording the root